src/wp-includes/admin-bar.php

```diff
@@ -1224,33 +1224,33 @@ function wp_admin_bar_add_secondary_groups( $wp_admin_bar ) {
 	);
 }
 
+
+
 /**
- * Prints style and scripts for the admin bar.
+ * Remove the `wp_admin_bar_header` action and use `wp_add_inline_style` function to render inline styles.
  *
- * @since 3.1.0
+ * @since 6.4.0
  */
-function wp_admin_bar_header() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?> media="print">#wpadminbar { display:none; }</style>
-	<?php
+function wp_admin_bar_header_styles() {
```

Comment:
A slightly better name:
```suggestion
function wp_enqueue_admin_bar_header_styles() {
```